xcm/xcm-builder/src/pay.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! `PayOverXcm` struct for paying through XCM and getting the status back
+
+use frame_support::traits::{
+	tokens::{Pay, PaymentStatus},
+	Get,
+};
+use polkadot_core_primitives::AccountId;
```

Comment:
This is a concrete type specialised for the Polkadot mainnet. Using it limits `PayOverXcm` to use only on Polkadot and other networks who share the exact same `AccountId` type. This is unacceptable.